### PR TITLE
Mnemonics

### LINKED ISSUE #38: Add support for base36 output of recovery seed

### PR DESCRIPTION
**Pull request by omartijn**
This adds word lists and the option to create and parse mnemonics fom and into recovery seeds.

Closes #38

### CODE REVIEW

File: master_key.h [hunk 1]
```diff
@@ -41,6 +61,290 @@ class master_key : public pgp::secure_object<std::array<uint8_t, crypto_kdf_KEYB
             crypto_kdf_keygen(data());
         }
 
+        /**
+         *  Query the user if they wish to use
+         *  encryption on their recovery seed
+         *
+         *  @return Whether or not to use encryption
+         */
+        static bool query_encrypt_seed()
```
Comment (bysnack) [inline master_key.h]: Couldn't this function be noexcept?

File: word_iterator.h [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        {
+            // check whether either the word or remaining data is different
+            return _word != that._word || _data != that._data;
+        }
+
+        /**
+         *  Retrieve the current word
+         */
+        constexpr const std::string_view &operator->() const noexcept { return _word; }
+        constexpr const std::string_view &operator*()  const noexcept { return _word; }
```
[the remark this answers is not in the document]
Comment (omartijn) [reply word_iterator.h]: Fair points. Changing the underlying view is, unfortunately, not possible. You can't change the contents of an `std::string_view`, it behaves like an `std::span<const char>` in that regard.

Perhaps a better name would be a `const_word_iterator`, given that the `const_` prefix is used in the Standard Library to indicate an iterator to something which cannot be changed.

[the remark this answers is not in the document]
Comment (bysnack) [reply word_iterator.h]: Yes, I get how string view works, I was talking about allowing the user to call the modifier member functions if they wanted such as `remove_prefix` and `swap`, that would make sense if we have a `word_view_iterator` but it does make sense to leave as it is if this is just a `const_word_iterator`.

Also, wouldn't it make sense to also update and use the traits for the return types if this is a `const iterator`?

[the remark this answers is not in the document]
Comment (omartijn) [reply word_iterator.h]: Yes, the traits were updated and are now used explicitly.

File: const_word_iterator.h [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ *  in a given string_view
+ */
+class const_word_iterator
+{
+    public:
+        /**
+         *  Iterator traits
+         */
+        using difference_type   = std::string_view::difference_type;
+        using value_type        = const std::string_view;
```
Comment (bysnack) [inline const_word_iterator.h]: The standard says:
screen

So shouldn't this be just `std::string_view`?

Comment (omartijn) [reply const_word_iterator.h]: Well, that seems strange to me, but we can't argue with the standard now, can we?

**Pull request merged**